WordPress/src/main/java/org/wordpress/android/ui/main/SitePickerSearchView.java

```diff
@@ -64,9 +63,7 @@ public boolean onMenuItemActionCollapse(MenuItem item) {
     public void enableSearchMode() {
         mMenuEdit.setVisible(false);
         mSitePickerActivity.setIsInSearchModeAndNullifyAdapter(true);
-        SitePickerSearchAdapter adapter = (SitePickerSearchAdapter) mSitePickerActivity.getAdapter();
-        mSitePickerActivity.getRecycleView().swapAdapter(adapter, true);
-        adapter.loadSites();
+        mSitePickerActivity.getRecycleView().swapAdapter(mSitePickerActivity.getAdapter(), true);
```

Comment:
> SitePickerSearchView was written so that all work related to search was contained in one place. That means it needs a reference to the Activity to do some work.

Couldn't this be done via a listener instead, which is passed by the activity to SitePickerSearchView? Something like this:

```
public interface OnSearchSitesListener {
        void onSearchSubmitted(String s);
        void onSearchEnabled();
        void onSearchDisabled();
}
```